# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatformFontList.h
# Commit: c03c7df7cd61
# Full Hash: c03c7df7cd61c88de130f49dbceaa8a62c9a7758
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-05-14 21:58:21
# Regressor Bug: 1708768
# File Overlap Count: 2
# Description:
#   Bug 1708768 - Preliminary cleanup of gfxPlatformFontList creation/initialization code; no functional change. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114863
# ==============================================================================

diff -r 76dd6379cc01 -r c03c7df7cd61 gfx/thebes/gfxPlatformFontList.h
--- a/gfx/thebes/gfxPlatformFontList.h	Fri May 14 17:45:16 2021 +0000
+++ b/gfx/thebes/gfxPlatformFontList.h	Fri May 14 18:05:44 2021 +0000
@@ -171,13 +171,13 @@
 
   static gfxPlatformFontList* PlatformFontList() { return sPlatformFontList; }
 
-  static nsresult Init() {
-    NS_ASSERTION(!sPlatformFontList, "What's this doing here?");
-    gfxPlatform::GetPlatform()->CreatePlatformFontList();
-    if (!sPlatformFontList) {
-      return NS_ERROR_OUT_OF_MEMORY;
+  static bool Initialize(gfxPlatformFontList* aList) {
+    sPlatformFontList = aList;
+    if (aList->InitFontList()) {
+      return true;
     }
-    return NS_OK;
+    Shutdown();
+    return false;
   }
 
   static void Shutdown() {
@@ -187,8 +187,8 @@
 
   virtual ~gfxPlatformFontList();
 
-  // initialize font lists
-  nsresult InitFontList();
+  // Initialize font lists; return true on success, false if something fails.
+  bool InitFontList();
 
   void FontListChanged();
 